index.js

```diff
@@ -42,7 +43,11 @@ WinstonDynamoDB.prototype.log = function (info, callback) {
     debug('log (called by winston)', info);
 
     if (!isEmpty(info.message) || isError(info.message)) {
-        this.add(info);
+        const { message } = info
+
+        for (let i = 0; i < message.length; i += maximalMessageLength) {
```

Comment:
I see this repo has the `popEventsToSend` function which truncated messages, char amount up to `MAX_EVENT_MSG_SIZE_BYTES`.

Seems like this code you added would make that code redundant.
Maybe your code is more fitting in that file?

Reply:
But after your changes, is there any chance the `popEventsToSend` will enter the if based on `MAX_EVENT_MSG_SIZE_BYTES`?
Isn't that dead code?

Reply:
I still don't think why it happen in the first place. shouldn't it be split anyway? IIUC your solution is splitting the string just before we insert, and then the already existing logic will probably won't catch it ( I guess that what @tomporat247 refer as dead code)

Reply:
> possible. 
10k chars => 80k bytes. 25 events => 2M bytes > 400k limit

the truncate is per event, I think we should delete that code if we go with yours, it is dead - the condition will always be false, no?

Reply:
ok, so we have 2 truncations. one is per message, the other for the entire batch. the entire batch, might be reached, the per event, not anymore. I'll remove just that, and revalidate all works in the other PR that relies on the new functionality.